### DISCUSSION TITLE
Support serving of the tracking script via reverse proxy

### DISCUSSION THREAD

**Discussion by bajtos** [plausible/analytics#437] [Anything Else]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the feature has not yet been requested.

## Feature request

**Is your feature request related to a problem? Please describe.**

I am looking for a way how to gather web analytics from Brave browser users. I am running my site on a custom domain https://bajtos.net/ and have configured Plausible to serve the tracking script at a custom domain [`s.bajtos.net`](https://s.bajtos.net/js/index.js). Unfortunately, Brave is blocking requests to `https://s.bajtos.net/js/index.js` - the Network tab shows error `(blocked:other)`. To workaround the issue, I created a rewrite/redirect rule in my Netlify config to "mount" all Plausible assets at `/stats` path (i.e. I am redirecting all `/stats/*` requests to `https://s.bajtos.net/:splat`). This works great for serving the initial `index.js` file. Unfortunately, the script then posts data to `https://s.bajtos.net/api/event`, which is blocked by the browser again :(

I am looking for a way how to configure Plausible to post events to `https://bajtos.net/stats/api/event` URL. 

**How would you like it to work**

Ideally, I would like the events API endpoint to be automatically inferred from the location of the main script. Is it possible to use a relative path, i.e. something like `../../api/event` instead of `https://s.bajtos.net/api/event`?

**Comment by metmarkosaric** [comment]
If you're concerned about adblockers, we now have instructions on how to proxy our script. See https://plausible.io/docs/proxy/introduction

**Reply by nicbou** [reply]
The instructions are missing the headers that are expected by Plausible, or concrete instructions for Nginx or Apache. I find myself piecing together bits of knowledge from random comments in the other discussion threads.

I lost 3 days of stats so far trying to get it to work properly. I only see 5% of my previous traffic, and no real explanation for it. It seems to work just fine on the client side, with a 202 response to the /event endpoint. Somehow, none of it gets tracked, and I can't figure out why.

**Reply by nicbou** [reply]
I have zero idea of what is happening with the rest of the traffic, but Google Search Console does not show any dip in search engine traffic. The request goes through just fine, the frontend cache is clear, and there is nothing unusual in the nginx logs, yet almost nothing gets logged. I'm truly at my wit's end.

```
    location = /whoisthere.js {
        proxy_pass https://stats.allaboutberlin.com/js/plausible.outbound-links.js;
        proxy_buffering on;
        proxy_cache jscache;
        proxy_cache_valid 200 6h;
        proxy_cache_use_stale updating error timeout invalid_header http_500;

        proxy_set_header Host stats.allaboutberlin.com;
        proxy_ssl_name stats.allaboutberlin.com;
        proxy_ssl_server_name on;
        proxy_ssl_session_reuse off;

        add_header X-Cache $upstream_cache_status;
    }

    location = /whoisthere/event {
        proxy_pass https://stats.allaboutberlin.com/api/event;
        proxy_buffering on;
        proxy_http_version 1.1;

        proxy_set_header Host stats.allaboutberlin.com;
        proxy_ssl_name stats.allaboutberlin.com;
        proxy_ssl_server_name on;
        proxy_ssl_session_reuse off;

        proxy_set_header X-Forwarded-For   $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host  $host;

        add_header X-Cache $upstream_cache_status;
    }
```

**Comment by bajtos** [comment]
I was able to successfully setup proxy-based integration of Plausible to my blog website, all is good 👍🏻 

I wrote down a how-to guide here: https://bajtos.net/posts/2022-01-privacy-friendly-web-analytics/
